src/api/app/models/staging/workflow.rb

```diff
@@ -57,12 +57,13 @@ def self.load_groups
     #
     # TODO: Refactor this code using to_h when updating to Ruby 2.6 (performance improvement)
     #
-    groups_hash = {}
-    # FIXME: try to use cache without failure. Like: Rails.cache.fetch(Group.all.cache_key) do
-    Group.find_each do |group|
-      groups_hash[group.title] = group
+    Rails.cache.fetch(Group.all.cache_key) do
```

Comment:
you have to make sure no other code is using this cache_key to cache another return value, so better append some code path specific key